pkg/appmanager/appManager.go

```diff
@@ -1978,6 +1990,95 @@ func (appMgr *Manager) recordIngressEvent(ing *v1beta1.Ingress,
 	appMgr.eventRecorder.Event(ing, v1.EventTypeNormal, reason, message)
 }
 
+// Resolve the first host name in an Ingress and use the IP address as the VS address
+func (appMgr *Manager) resolveIngressHost(ing *v1beta1.Ingress, namespace string) {
+	var host, ipAddress string
+	var err error
+	var netIPs []net.IP
+	logDNSError := func(msg string) {
+		log.Warning(msg)
+		appMgr.recordIngressEvent(ing, "DNSResolutionError", msg, "")
+	}
+
+	if nil != ing.Spec.Rules {
+		// Use the host from the first rule
+		host = ing.Spec.Rules[0].Host
+		if host == "" {
+			// Host field is empty
+			logDNSError(fmt.Sprintf("First host is empty on Ingress '%s'; cannot resolve.",
+				ing.ObjectMeta.Name))
+			return
+		}
+	} else {
+		logDNSError(fmt.Sprintf("No host found for DNS resolution on Ingress '%s'",
+			ing.ObjectMeta.Name))
+		return
+	}
+
+	if appMgr.resolveIng == "LOOKUP" {
+		// Use local DNS
+		netIPs, err = net.LookupIP(host)
+		if nil != err {
+			logDNSError(fmt.Sprintf("Error while resolving host '%s': %s", host, err))
+			return
+		} else {
+			if len(netIPs) > 1 {
+				log.Warningf(
+					"Resolved multiple IP addresses for host '%s', "+
+						"choosing first resolved address.", host)
+			}
+			ipAddress = netIPs[0].String()
+		}
+	} else {
+		// Use custom DNS server
+		customDNS := appMgr.resolveIng
+		isIP := net.ParseIP(customDNS)
+		if isIP == nil {
+			// customDNS is not an IPAddress, it is a hostname that we need to resolve first
+			netIPs, err = net.LookupIP(customDNS)
+			if nil != err {
+				logDNSError(fmt.Sprintf("Error while resolving host '%s': %s",
+					appMgr.resolveIng, err))
+				return
+			}
+			customDNS = netIPs[0].String()
+		}
+		client := dns.Client{}
+		msg := dns.Msg{}
+		msg.SetQuestion(host+".", dns.TypeA)
+		res, _, err := client.Exchange(&msg, customDNS+":53")
```

Comment:
It is possible that the custom DNS server may be running on a different port than 53, so we should handle that case by only appending ':53' in the case where customDNS does not already contain a port.